### PR TITLE
libtiff: bump to libtiff/4.3.0

### PR DESCRIPTION
**Pull request by madebr**
Specify library name and version:  **libtiff/4.3.0**

bump

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by SSE4 — DISMISSED** [review]

File: recipes/libtiff/all/conanfile.py [hunk 1]
```diff
@@ -120,7 +122,27 @@ def _patch_sources(self):
         tools.replace_in_file(cmakefile,
                               "add_subdirectory(tools)\nadd_subdirectory(test)\nadd_subdirectory(contrib)\nadd_subdirectory(build)\n"
                               "add_subdirectory(man)\nadd_subdirectory(html)", "")
-        tools.replace_in_file(cmakefile, "LIBLZMA_LIBRARIES", "LibLZMA_LIBRARIES")
+        if tools.Version(self.version) < "4.3.0":
+            tools.replace_in_file(cmakefile, "LIBLZMA_LIBRARIES", "LibLZMA_LIBRARIES")
```
Comment (SpaceIm) [inline recipes/libtiff/all/conanfile.py:126]: Not required anymore, `xz_utils` recipe has been fixed in https://github.com/conan-io/conan-center-index/pull/5231

Comment (prince-chrismc) [reply recipes/libtiff/all/conanfile.py:126]:
```suggestion
```

File: recipes/libtiff/all/conanfile.py [hunk 2]
```diff
@@ -120,7 +122,27 @@ def _patch_sources(self):
...
+        if tools.Version(self.version) >= "4.3.0":
+            if self.options.get_safe("jbig"):
+                tools.save("Findjbig.cmake", textwrap.dedent("""\
```
Comment (prince-chrismc) [inline recipes/libtiff/all/conanfile.py:129]: This

File: recipes/libtiff/all/conanfile.py [hunk 3]
```diff
@@ -120,7 +122,27 @@ def _patch_sources(self):
...
+                    if(NOT TARGET JBIG::JBIG)
+                        add_library(JBIG::JBIG INTERFACE IMPORTED)
+                        target_link_libraries(JBIG::JBIG INTERFACE jbig::jbig)
+                    endif()
+                """), append=True)
+            if self.options.get_safe("zstd"):
+                tools.save("Findjbig.cmake", textwrap.dedent("""\
```
Comment (prince-chrismc) [inline recipes/libtiff/all/conanfile.py:136]: vs this

Comment (prince-chrismc) [inline recipes/libtiff/all/conanfile.py:136]: Copy paste error?

```suggestion
                tools.save("Findzstd.cmake", textwrap.dedent("""\
```

File: recipes/libtiff/all/patches/4.3.0-0001-cmake-add-libjpeg-turbo.patch [hunk 4]
```diff
@@ -0,0 +1,50 @@
+--- cmake/FindZSTD.cmake
++++ cmake/FindZSTD.cmake
+@@ -1,0 +1,9 @@
++include("${cmake_wrapper_SOURCE_DIR}/Findzstd.cmake")
++if(NOT TARGET ZSTD::ZSTD)
++  add_library(ZSTD::ZSTD INTERFACE IMPORTED)
++  if(TARGET zstd::libzstd_shared)
++    target_link_libraries(ZSTD::ZSTD INTERFACE zstd::libzstd_shared)
```
Comment (prince-chrismc) [inline recipes/libtiff/all/patches/4.3.0-0001-cmake-add-libjpeg-turbo.patch:8]: Seems odd to be making the same targets that were saved in the recipe 😕

File: recipes/libtiff/all/patches/4.3.0-0001-cmake-add-libjpeg-turbo.patch [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
++    target_link_libraries(ZSTD::ZSTD INTERFACE zstd::libzstd_static)
++  endif()
++endif()
+--- cmake/FindJBIG.cmake
++++ cmake/FindJBIG.cmake
+@@ -1,0 +1,5 @@
++include("${cmake_wrapper_SOURCE_DIR}/Findjbig.cmake")
++if(NOT TARGET JBIG::JBIG)
++  add_library(JBIG::JBIG INTERFACE IMPORTED)
++  target_link_libraries(JBIG::JBIG INTERFACE jbig::jbig)
```
Comment (prince-chrismc) [inline recipes/libtiff/all/patches/4.3.0-0001-cmake-add-libjpeg-turbo.patch:19]: Again this looks duplicated in the recipe

**Review by prince-chrismc — COMMENTED** [review]
I am confused by the duplication in the package and the generated files from the recipes....

Would it be more clean to move those generated blips to separate files and export them? would clean up the recipe

File: recipes/libtiff/all/conanfile.py (hunk 3 above)
Comment (madebr) [reply recipes/libtiff/all/conanfile.py:136]: This should be `FindZSTD.cmake`.

Comment (madebr) [reply recipes/libtiff/all/conanfile.py:136]: Yup, you have eagle eyes.

File: recipes/libtiff/all/patches/4.3.0-0001-cmake-add-libjpeg-turbo.patch (hunk 5 above)
Comment (madebr) [reply recipes/libtiff/all/patches/4.3.0-0001-cmake-add-libjpeg-turbo.patch:19]: Indeed, I removed the `tools.replace_in_file`'s in the conanfile.
Everything is now handled by a patch.

**Review by prince-chrismc — APPROVED** [review]
LGTM! 🚀

File: recipes/libtiff/all/conanfile.py [hunk 6]
```diff
@@ -1,6 +1,9 @@
 from conans import ConanFile, CMake, tools
 from conans.errors import ConanInvalidConfiguration
 import os
+import textwrap
```
Comment (Nekto89) [inline recipes/libtiff/all/conanfile.py:4]:
```suggestion
```
not used anymore

File: recipes/libtiff/all/conandata.yml [hunk 7]
```diff
@@ -12,6 +15,9 @@ sources:
     url: "http://download.osgeo.org/libtiff/tiff-4.0.8.tar.gz"
     sha256: "59d7a5a8ccd92059913f246877db95a2918e6c04fb9d43fd74e5c3390dac2910"
 patches:
+  "4.3.0":
+    - patch_file: "patches/4.3.0-0001-cmake-add-libjpeg-turbo.patch"
```
Comment (Nekto89) [inline recipes/libtiff/all/conandata.yml:19]: maybe rename file? It does more than just adding libjpeg-turbo

File: recipes/libtiff/all/conanfile.py [hunk 8]
```diff
@@ -96,11 +96,20 @@ def requirements(self):
...
-        tools.get(**self.conan_data["sources"][self.version])
-        os.rename("tiff-" + self.version, self._source_subfolder)
+        tools.get(**self.conan_data["sources"][self.version],
+                  destination=self._source_subfolder, strip_root=True)
 
     def _patch_sources(self):
+        # Rename the generated Findjbig.cmake and Findzstd.cmake to avoid case insensitive conflicts with FindJBIG.cmake and FindZSTD.cmake on Windows
+        tools.rename(os.path.join(self.build_folder, "Findjbig.cmake"),
+                     os.path.join(self.build_folder, "ConanFindjbig.cmake"))
+        tools.rename(os.path.join(self.build_folder, "Findzstd.cmake"),
```
Comment (uilianries) [inline recipes/libtiff/all/conanfile.py:111]:
```suggestion
        if tools.Version(self.version) >= "4.1":
            tools.rename(os.path.join(self.build_folder, "Findzstd.cmake"),
```

File: recipes/libtiff/all/conanfile.py [hunk 9]
```diff
@@ -96,11 +96,20 @@ def requirements(self):
...
+                     os.path.join(self.build_folder, "ConanFindzstd.cmake"))
```
Comment (madebr) [inline recipes/libtiff/all/conanfile.py:112]:
```suggestion
        if tools.Version(self.version) >= "4.1":
            tools.rename(os.path.join(self.build_folder, "Findzstd.cmake"),
                         os.path.join(self.build_folder, "ConanFindzstd.cmake"))
```

File: recipes/libtiff/all/conanfile.py (hunk 8 above)
Comment (madebr) [reply recipes/libtiff/all/conanfile.py]: Thanks! I've added your suggestion but with the next line also indented extra.

File: recipes/libtiff/all/conanfile.py [hunk 10]
```diff
@@ -132,7 +141,8 @@ def _configure_cmake(self):
             if self._has_libdeflate_option:
                 self._cmake.definitions["libdeflate"] = self.options.libdeflate
                 if self.options.libdeflate:
-                    self._cmake.definitions["DEFLATE_NAMES"] = self.deps_cpp_info["libdeflate"].libs[0]
+                    if tools.Version(self.version) < "4.3.0":
+                        self._cmake.definitions["DEFLATE_NAMES"] = self.deps_cpp_info["libdeflate"].libs[0]
```
Comment (SpaceIm) [inline recipes/libtiff/all/conanfile.py:145]: How libdeflate is discovered by libtiff in 4.3.0?

**Comment by SpaceIm** [comment]
@madebr In the meantime, could you bump libdeflate please?

**Review by SpaceIm — DISMISSED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Review by SpaceIm — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Pull request merged**